src/git-kill

```diff
@@ -13,10 +13,11 @@ function branch_to_kill {
 
 function preconditions {
   target_branch=$(branch_to_kill "$@")
-  parent_branch_name=$(parent_branch "$target_branch")
-
   ensure_is_feature_branch "$target_branch" "You can only kill feature branches."
 
+  ensure_knows_parent_branches "$target_branch"
```

Comment:
The plan is to add feature tests for each command that needs it into here. Planning on moving the git sync, and git ship tests in here as well